sklearn/utils/_array_api.py

```diff
@@ -108,17 +116,21 @@ def _check_array_api_dispatch(array_api_dispatch):
                 f"NumPy must be {min_numpy_version} or newer to dispatch array using"
                 " the API specification"
             )
-        if os.environ.get("SCIPY_ARRAY_API") != "1":
-            warnings.warn(
+
+        if not scipy._lib._array_api.SCIPY_ARRAY_API:
```

Comment:
Since is this private API I would do something link:

```python
try:
   scipy_array_api_enabled = scipy._lib._array_api.SCIPY_ARRAY_API
except AttributeError:
   warnings.warn(
       "Could not inspect if scipy array API support is enabled or not, assuming it is."
   )
   scipy_array_api_enabled = True

if not scipy_array_api_enabled:
   message = ...
```

Otherwise, a future version of scipy where array API support is enabled by default and does not define this private attribute anymore might break scikit-learn.